### PR TITLE
Build settings JAR from contents

### PR DESCRIPTION
**Pull request by ronocod**
Replaces the binary JAR for Android Studio settings with the contents of the file and a shell script to build the JAR. This means we get a proper diff when we update the settings.

We can add a CI job once this is merged to build the new JAR automatically when changes are made.

If you want to edit the settings from the IDE and merge your changes in you can do it like this:
1 - Import the JAR from this repository.
2 - Change your settings in the IDE
3 - Export your JAR to `/ide-settings/intellij/settings/settings.jar` in this repo
4 - Run `unzip /ide-settings/intellij/settings/settings.jar` from the repo base directory
5 - Delete `/ide-settings/intellij/settings/settings.jar`

### CODE REVIEW

**Comment by ataulm** [comment]
:heart:

**Comment by ataulm** [comment]
This means Novoda/Project/Individual settings are starting to become a viable thing!

File: ide-settings/intellij/settings/fileTemplates/includes/File Header.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/**
+ * User: ${USER}
+ * Date: ${DATE}
+ * Time: ${TIME}
```
Comment (ataulm) [inline ide-settings/intellij/settings/fileTemplates/includes/File Header.java]: This is added to the top of files? I thought the desired template is to not have it?

File: ide-settings/intellij/settings/fileTemplates/j2ee/Activity.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+import android.app.Activity;
+import android.os.Bundle;
+
+
+public class ${NAME} extends Activity {
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+}
```
Comment (ataulm) [inline ide-settings/intellij/settings/fileTemplates/j2ee/Activity.java]: can you change all the templates so that there's a blank line at end of file?

Comment (ouchadam) [inline ide-settings/intellij/settings/fileTemplates/j2ee/Activity.java]: should be able to auto format all the files

File: ide-settings/intellij/settings/fileTemplates/includes/File Header.java (hunk 1 above)
Comment (ronocod) [inline ide-settings/intellij/settings/fileTemplates/includes/File Header.java]: I didn't think this was in the JAR, but all the files here are just what's unzipped from the existing JAR, I didn't make any changes.

**Comment by ouchadam** [comment]
:100: from me, do we need a job to build the .jar?

File: ide-settings/intellij/settings/options/editor.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<application>
+  <component name="EditorSettings">
+    <option name="USE_SOFT_WRAPS" value="CONSOLE" />
+    <option name="IS_VIRTUAL_SPACE" value="true" />
+    <option name="IS_ENSURE_NEWLINE_AT_EOF" value="true" />
+    <option name="QUICK_DOC_ON_MOUSE_OVER_DELAY_MS" value="400" />
+    <option name="ARE_LINE_NUMBERS_SHOWN" value="true" />
+    <option name="IS_CAMEL_WORDS" value="true" />
```
Comment (amlcurran) [inline ide-settings/intellij/settings/options/editor.xml]: whilst we're here, can we take this one out :P

**Comment by ataulm** [comment]
@ouchadam nice idea, it can push back to the repo (unless that kicks off another build...)

**Comment by ronocod** [comment]
Can we have a job that makes GitHub releases with the compiled JAR? Then we can remove the JAR from the repo itself

File: ide-settings/intellij/settings/options/editor.xml (hunk 3 above)
Comment (ataulm) [inline ide-settings/intellij/settings/options/editor.xml]: :100: but @blundell and @ouchadam are odd people and like it

Comment (ronocod) [inline ide-settings/intellij/settings/options/editor.xml]: I'm all in favour, that can be another PR though :smile:

Comment (ataulm) [inline ide-settings/intellij/settings/options/editor.xml]: @amlcurran this is a perfect use case for implementing overriding values for individuals. E.g. a different directory with individual (/project) settings that will be used if available, only overriding the ones that are there I mean. Like merging.

Comment (stefanhoth) [inline ide-settings/intellij/settings/options/editor.xml]: Re: github releases. On another project we just set up a github upload so I
think it should be quick to add this here as well.
On Oct 9, 2015 16:39, "Ataul Munim" <notifications@github.com> wrote:

> In ide-settings/intellij/settings/options/editor.xml
> <https://github.com/novoda/novoda/pull/15#discussion_r41636946>:
>
> > @@ -0,0 +1,13 @@
> > +<application>
> > +  <component name="EditorSettings">
> > +    <option name="USE_SOFT_WRAPS" value="CONSOLE" />
> > +    <option name="IS_VIRTUAL_SPACE" value="true" />
> > +    <option name="IS_ENSURE_NEWLINE_AT_EOF" value="true" />
> > +    <option name="QUICK_DOC_ON_MOUSE_OVER_DELAY_MS" value="400" />
> > +    <option name="ARE_LINE_NUMBERS_SHOWN" value="true" />
> > +    <option name="IS_CAMEL_WORDS" value="true" />
>
> @amlcurran <https://github.com/amlcurran> this is a perfect use case for
> implementing overriding values for individuals. E.g. a different directory
> with individual (/project) settings that will be used if available, only
> overriding the ones that are there I mean. Like merging.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/novoda/novoda/pull/15/files#r41636946>.
>

-- 
Disclaimer: The information in this e-mail and any attachments is the 
property of Novoda Ltd and is confidential and may be legally privileged. 
This e-mail is intended solely for the person or organisation to which it 
is addressed. Any disclosure, copying or other use of the information by 
any person or organisation who is not the intended recipient is strictly 
prohibited and may be unlawful. If you have received this e-mail in error, 
please inform the sender immediately and delete/destroy this e-mail and any 
copies of it. Novoda Ltd has taken reasonable precautions to minimise the 
risk of any software viruses which may damage your systems, but we advise 
that you take the necessary steps to ensure that no virus contamination is 
suffered. Novoda Ltd does not accept any liability for any loss or damage 
caused by the transmission of any virus.

Novoda Ltd, Company No: 347444, Registered in Scotland Registered 
Office: C/O Alexander Sloan, 38 Cadogan Street, Glasgow, G2 7HF, Scotland.  VAT 
Registration Number  GB 984 2525 93

Comment (blundell) [inline ide-settings/intellij/settings/options/editor.xml]: Yeah this is the default setting, so @amlcurran new starters should have this setting, once proficient they can change it, i.e. you can change your local setting :-)

**Comment by blundell** [comment]
:100:

File: ide-settings/intellij/settings/fileTemplates/j2ee/Activity.java (hunk 2 above)
Comment (rock3r) [inline ide-settings/intellij/settings/fileTemplates/j2ee/Activity.java]: There's a setting for that in the Editor settings, no need for manually doing it.

Comment (ataulm) [inline ide-settings/intellij/settings/fileTemplates/j2ee/Activity.java]: yeah I meant to change that setting in Editor settings and reexport this set of config files :dancer:

File: ide-settings/intellij/settings/options/editor.xml (hunk 3 above)
Comment (rock3r) [inline ide-settings/intellij/settings/options/editor.xml]: What about having it off by default, so when they get proficient they can enable it?

File: ide-settings/intellij/settings/fileTemplates/j2ee/Activity.java (hunk 2 above)
Comment (ataulm) [inline ide-settings/intellij/settings/fileTemplates/j2ee/Activity.java]: no I didn't. I meant on the template. but I think the Editor setting should be better probably.

File: ide-settings/intellij/settings/options/editor.xml (hunk 3 above)
Comment (ataulm) [inline ide-settings/intellij/settings/options/editor.xml]: nooo leave it on - this is going to be the guinea pig on how easy it is to customise your own settings on top of the novoda settings, while still keeping it easy to update novoda settings.

**Comment by ronocod** [comment]
No comments for a few days, do we want to :ship: this or is there more work to do before then?

**Comment by rock3r** [comment]
LGTM, will probably tweak a few things once this is merged :)
@stefanhoth you want to contribute a Windows Shell version?

**Comment by stefanhoth** [comment]
@rock3r yeah, it's almost the same but I can't do it today.

**Comment by rock3r** [comment]
@stefanhoth can we ship this in the meantime? Maybe do the Windows part in a separate PR.

**Comment by rock3r** [comment]
Let's :ship: :it: thennnnn

**Comment by blundell** [comment]
:scream:

**Comment by rock3r** [comment]
Yayssss thanks @blundell :)
